### PR TITLE
[dev] Add source archives to gitignore

### PR DESCRIPTION
**Pull request by thcrain-msft**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

We have a policy of not checking in source archives to this repo, for good reason. This PR partially manifests that policy in our gitignore file. I've found this to be a small workflow improvement for myself, and I believe this will also help first-time contributors as well.

###### Change Log  

- Add source archives to gitignore

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Been using these patterns in my personal `.git/info/excludes` file

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,11 @@
 ##
 ## Get latest from https://github.com/github/gitignore/blob/master/VisualStudio.gitignore
 
+# Exclude source archives from the spec directory
+SPECS/**/*.tar.*
+SPECS/**/*.tgz
+SPECS/**/*.zip
```
Comment (PawelWMS) [inline .gitignore:9]: Good idea! Could you also remove any existing archives, which accidentally got into our repo as well? I was able to find only two right now:
- `SPECS/telegraf/telegraf-vendor-1.14.5.tar.gz` and
- `SPECS/zipper/minizip.tar.gz`.

```suggestion
SPECS/**/*.xz
SPECS/**/*.zip
```

Comment (PawelWMS) [reply .gitignore:9]: We could also [embrace what others have already came up with in this space](https://github.com/github/gitignore/blob/master/Global/Archives.gitignore).

Comment (thcrain-msft) [reply .gitignore:9]: Oh awesome, that list is way more thorough. I'll go ahead and implement that instead. Will remove existing archives as well (assuming they're not load-bearing mistakes, somehow).

Comment (thcrain-msft) [reply .gitignore]: Removed the existing archives, and checked that they're in the source blob.

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,32 @@
...
+SPECS/**/*.dmg
+SPECS/**/*.xpi
+SPECS/**/*.gem
+SPECS/**/*.egg
+SPECS/**/*.deb
+SPECS/**/*.rpm
+SPECS/**/*.msi
+SPECS/**/*.msm
+SPECS/**/*.msp
+SPECS/**/*.txz
```
Comment (PawelWMS) [inline .gitignore:28]: Looks good, let's just put it in alphabetical order:
```suggestion
SPECS/**/*.7z
SPECS/**/*.bz2
SPECS/**/*.bzip
SPECS/**/*.bzip2
SPECS/**/*.cab
SPECS/**/*.deb
SPECS/**/*.dmg
SPECS/**/*.egg
SPECS/**/*.gem
SPECS/**/*.gz
SPECS/**/*.gzip
SPECS/**/*.iso
SPECS/**/*.jar
SPECS/**/*.lzma
SPECS/**/*.msi
SPECS/**/*.msm
SPECS/**/*.msp
SPECS/**/*.rar
SPECS/**/*.rpm
SPECS/**/*.tar
SPECS/**/*.tgz
SPECS/**/*.txz
SPECS/**/*.xar
SPECS/**/*.xpi
SPECS/**/*.xz
```

**Review by PawelWMS — APPROVED** [review]

**Pull request merged**